lib/command.c

```diff
@@ -470,8 +487,8 @@ static char *zencrypt(const char *passwd)
 /* This function write configuration of this host. */
 static int config_write_host(struct vty *vty)
 {
-	if (host.name)
-		vty_out(vty, "hostname %s\n", host.name);
+	if (hostname_get())
+		vty_out(vty, "hostname %s\n", hostname_get());
```

Comment:
would be good to extend this so we can set the domainname in the config, especially for the platforms that don't return a domainname in uname. Could be added (and saved) as a second optional parameter on "hostname", e.g. `"hostname HOSTNAME [DOMAINNAME]"`. (Or alternatively a new `domainname` CLI command, either is fine I guess.)

Reply:
Agree with @donaldsharp. We should probably open a bug for this and try to solve it separately.

Reply:
Yes this is adding a new feature here... by adding tracking of the domain name in general. I.e. I see this as your change already adding a new feature, but it's incomplete -- there is no way to get a domain name on any of the BSDs or Solaris.

To me, this is really just half done; and since it's not a massively complicated thing I'd prefer to not "half-add" it.

Also, ... I would like to point out that we're bikeshedding on a hostname thing on a PR that largely contains EVPN fixes. I'm fully aware this is how things work out in development, but I'd still like to note that if this was 2 separate PRs, the EVPN bits would've been in a week ago...

Reply:
P.S.: I noticed that at least on OpenBSD 6.0, I get the entire FQDN in `names.nodename`.  The code should probably split it on the first dot?

Reply:
Made the changes. Added a command to set domainname as well. We will user configured hostname/domainname when configured but use system info when the user config is not present.